test/content/support.js

```diff
@@ -167,4 +196,308 @@ function resetDB() {
+			if (json === null) json = indent + '\tnull'; // function
+			str += '\n' + json + (i < obj.length-1 ? ',' : '');
+		}
+		return str + (obj.length ? '\n' + indent : '') + ']';
+	}
+	
+	let keys = Object.keys(obj).sort(),
+		empty = true,
+		str = indent + label + '{';
+	for (let i=0; i<keys.length; i++) {
+		let json = stableStringify(obj[keys[i]], level + 1, keys[i]);
+		if (json === null) continue; // function
+		
+		empty = false;
+		str += '\n' + json + (i < keys.length-1 ? ',' : '');
+	}
+	
+	return str + (!empty ? '\n' + indent : '') + '}';
+}
+
+/**
+ * Generates sample item data that is stored in data/sampleItemData.js
+ */
+function generateAllTypesAndFieldsData() {
+	let data = {};
+	let itemTypes = Zotero.ItemTypes.getTypes();
+	// For most fields, use the field name as the value, but this doesn't
+	// work well for some fields that expect values in certain formats
+	let specialValues = {
+		date: '1999-12-31',
+		filingDate: '2000-01-02',
+		accessDate: '1997-06-13 23:59:58',
+		number: 3,
+		numPages: 4,
+		issue: 5,
+		volume: 6,
+		numberOfVolumes: 7,
+		edition: 8,
+		seriesNumber: 9,
+		ISBN: '978-1-234-56789-7',
+		ISSN: '1234-5679',
+		url: 'http://www.example.com',
+		pages: '1-10',
+		DOI: '10.1234/example.doi',
+		runningTime: '1:22:33',
+		language: 'en-US'
+	};
+	
+	// Item types that should not be included in sample data
+	let excludeItemTypes = ['note', 'attachment'];
+	
+	for (let i = 0; i < itemTypes.length; i++) {
+		if (excludeItemTypes.indexOf(itemTypes[i].name) != -1) continue;
+		
+		let itemFields = data[itemTypes[i].name] = {
+			itemType: itemTypes[i].name
+		};
+		
+		let fields = Zotero.ItemFields.getItemTypeFields(itemTypes[i].id);
+		for (let j = 0; j < fields.length; j++) {
+			let field = fields[j];
+			field = Zotero.ItemFields.getBaseIDFromTypeAndField(itemTypes[i].id, field) || field;
+			
+			let name = Zotero.ItemFields.getName(field),
+				value;
+			
+			// Use field name as field value
+			if (specialValues[name]) {
+				value = specialValues[name];
+			} else {
+				value = name.charAt(0).toUpperCase() + name.substr(1);
+				// Make it look nice (sentence case)
+				value = value.replace(/([a-z])([A-Z])/g, '$1 $2')
+					.replace(/ [A-Z](?![A-Z])/g, m => m.toLowerCase()); // not all-caps words
+			}
+			
+			itemFields[name] = value;
+		}
+		
+		let creatorTypes = Zotero.CreatorTypes.getTypesForItemType(itemTypes[i].id),
+			creators = itemFields.creators = [];
+		for (let j = 0; j < creatorTypes.length; j++) {
+			let typeName = creatorTypes[j].name;
+			creators.push({
+				creatorType: typeName,
+				firstName: typeName + 'First',
+				lastName: typeName + 'Last'
+			});
+		}
+	}
+	
+	return data;
+}
+
+/**
+ * Loads specified sample data from file
+ */
+function loadSampleData(dataName) {
+	Components.utils.import("resource://gre/modules/Services.jsm");
+	let data = {};
+	Services.scriptloader.loadSubScript('resource://zotero-unit-tests/data/' + dataName + '.js', data, 'UTF-8');
```

Comment:
Is there a reason we need to use the subscript loader here rather than loading as JSON?